⚠️ Align KubeadmConfig to kubeadm v1beta4 types (#12282)
kubernetes-sigs/cluster-api: docs/book/src/developer/providers/migrations/v1.10-to-v1.11.md

- KubeadmConfig (and the entire CABPK provider) now implements the v1beta2 Cluster API contract.
- See changes that apply to [all CRDs](#all-crds)
- ExtraArg field types have been changed from `map[string]sting` to `[]Arg`, thus aligning with kubeadm v1beta4 API;

>   however, using multiple args with the same name will be enabled only when v1beta1 is removed, tentatively tentatively in August 2026

Reviewer: ```suggestion
  however, using multiple args with the same name will be enabled only when v1beta1 is removed, tentatively in August 2026
```

  - `spec.clusterConfiguration.apiServer.extraArgs` type has been changed to `[]Arg`
  - `spec.clusterConfiguration.controllerManager.extraArgs` type has been changed to `[]Arg`
  - `spec.clusterConfiguration.scheduler.extraArgs` type has been changed to `[]Arg`

>   - `spec.clusterConfiguration.etcd.local.nodeRegistration.extraArgs` type has been changed to `[]Arg`

Reviewer: ```suggestion
  - `spec.clusterConfiguration.etcd.local.extraArgs` type has been changed to `[]Arg`
```

  - `spec.initConfiguration.nodeRegistration.kubeletExtraArgs` type has been changed to `[]Arg`
  - `spec.joinConfiguration.nodeRegistration.kubeletExtraArgs` type has been changed to `[]Arg`
- ImagePullPolicy field types have been changed from `string` to `corev1.PullPolicy`, thus aligning with kubeadm v1beta4 API
  - `spec.initConfiguration.nodeRegistration.imagePullPolicy` type has been changed to `corev1.PullPolicy`
  - `spec.joinConfiguration.nodeRegistration.imagePullPolicy` type has been changed to `corev1.PullPolicy`

> - Timeout fields have been aligned with kubeadm v1beta4 API, but filed name and type has been adapted according to API guidelines.

Reviewer: ```suggestion
- Timeout fields have been aligned with kubeadm v1beta4 API, but field names and types have been adapted according to API guidelines.
```

  - `spec.initConfiguration.timeouts` struct has been added with the following fields:
    - `controlPlaneComponentHealthCheckSeconds`
    - `kubeletHealthCheckSeconds`
    - `kubernetesAPICallSeconds`
    - `etcdAPICallSeconds`
    - `tlsBootstrapSeconds`
    - `discoverySeconds`
  - `spec.joinConfiguration.timeouts` field has been added with the same set of timeouts listed above.

>   - `spec.clusterConfiguration.apiServer.timeoutForControlPlane` field has been removed. Use `spec.initConfiguration.timeouts.controlPlaneComponentHealthCheckSeconds` instead.

Reviewer: also joinConfiguration?
